cmd/modelcmd/controller.go

```diff
@@ -109,9 +110,18 @@ func (c *ControllerCommandBase) initController() error {
 // 2. if JUJU_CONTROLLER env variable exists, its value;
 // 3. current controller specified in local controllers.yaml file.
 func DetermineCurrentController(store jujuclient.ClientStore) (string, error) {
-	controllerName, _ := SplitModelName(os.Getenv(osenv.JujuModelEnvKey))
+	modelController, _ := SplitModelName(os.Getenv(osenv.JujuModelEnvKey))
+	envController := os.Getenv(osenv.JujuControllerEnvKey)
+	if modelController != "" && envController != "" && modelController != envController {
+		return "", errors.Trace(errors.New(
```

Comment:
You don't need to Trace() a New() error, as it already has the file and line information.
We also have
errors.Errorf() that lets you put the sprintf in directly, so this can be simplified to just

return "", errors.Errorf("controller name from ...